src/core/Song.cpp

```diff
@@ -266,13 +266,22 @@ void Song::processNextBuffer()
 	}
 
 	f_cnt_t framesPlayed = 0;
+	f_cnt_t framesToPlay = Engine::mixer()->framesPerPeriod() - framesPlayed;
+
+	float currentFrame = m_playPos[m_playMode].currentFrame();
+	if( ( f_cnt_t ) currentFrame == 0 )
+	{
+		processAutomations(trackList, m_playPos[m_playMode], framesToPlay);
```

Comment:
Well, it works. I'm still reading up on `Song::processAutomations( ... )` though.